### PR TITLE
Big code reorg

### PR DESCRIPTION
**Pull request by jeromekelleher**
Depends on #989 

Trying to make the code layout make sense, particularly getting ready for adding lots of stuff in the mutations module.

Nuked a bunch of ancient testing code, which was *very* satisfying.

### CODE REVIEW

**Comment by jeromekelleher** [comment]
This is ready for review now I think - good to see what people think of the new layout. Is ``ancestry.py`` the right name for the module where we do the tree simulations?

File: msprime/demography.py [hunk 1]
```diff
@@ -0,0 +1,1091 @@
...
+        return {
+            "type": "migration_rate_change",
+            "time": self.time,
+            "migration_rate": self.rate,
+            "source": self.source,
+            "dest": self.dest,
+        }
+
+    def __str__(self):
+        if self.source == -1 is self.dest == -1:
```
Comment (benjeffery) [inline msprime/demography.py:204]: I know this isn't new code, just moved, but current line gives `SyntaxWarning` on py 3.8.
```suggestion
        if (self.source == -1) == (self.dest == -1)
```

Comment (jeromekelleher) [reply msprime/demography.py]: Eek! Not sure how that one snuck through.

File: msprime/demography.py [hunk 2]
```diff
@@ -0,0 +1,1091 @@
...
+        return {
+            "type": "migration_rate_change",
+            "time": self.time,
+            "migration_rate": self.rate,
+            "source": self.source,
+            "dest": self.dest,
+        }
+
+    def __str__(self):
+        if (self.source == -1) == (self.dest == -1)
```
Comment (benjeffery) [inline msprime/demography.py:204]: Sorry my last suggested commit contained a typo!
```suggestion
        if (self.source == -1) == (self.dest == -1):
```

**Comment by jeromekelleher** [comment]
Thanks @benjeffery - I'll tidy this up and merge soon.

**Comment by petrelharp** [comment]
Looks good to me, also!

**Pull request merged**